modules/instrument_manager/jsx/instrumentManagerIndex.js

```diff
@@ -0,0 +1,174 @@
+    this.formatColumn = this.formatColumn.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   * Additionally add hiddenHeaders to global loris variable
+   * for easy access by columnFormatter.
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL, {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({data}))
+      .catch((error) => {
+        this.setState({error: true});
+        console.error(error);
+      });
+  }
+
+  /**
+   * Modify behaviour of specified column cells in the Data Table component
+   *
+   * @param {string} column - column name
+   * @param {string} cell - cell content
+   * @param {object} row - row content indexed by column
+   *
+   * @return {*} a formated table cell for a given column
+   */
+  formatColumn(column, cell, row) {
+    // Set class to 'bg-danger' if file is hidden.
+    return (
+      <td>{cell}</td>
+    );
+  }
+
+  render() {
+    // If error occurs, return a message.
+    // XXX: Replace this with a UI component for 500 errors.
+    if (this.state.error) {
+      return <h3>An error occured while loading the page.</h3>;
+    }
+
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return <Loader/>;
+    }
+
+    const fields = [
+      {label: 'Instrument', show: true, filter: {
+        name: 'instrument',
+        type: 'text',
+      }},
+      {label: 'Instrument Type', show: true, filter: {
+        name: 'instrument_type',
+        type: 'select',
+        options: {
+          'Instrument Builder': 'Instrument Builder',
+          'PHP': 'PHP',
+          'Missing': 'Missing',
+        },
+      }},
+      {label: 'Table Installed', show: true, filter: {
+        name: 'table_installed',
+        type: 'select',
+        options: {
+          'Exists': 'Exists',
+          'Missing': 'Missing',
+        },
+      }},
+      {label: 'Table Valid', show: true, filter: {
+        name: 'table_valid',
+        type: 'text',
+      }},
+      {label: 'Pages Valid', show: true, filter: {
+        name: 'pages_valid',
+        type: 'text',
+      }},
+    ];
+
+    const tabs = [
+      {id: 'browse', label: 'Browse'},
+      {id: 'upload', label: 'Upload'},
+    ];
+
+    const feedback = () => {
+      if (!this.state.data.caninstall) {
+        return (
+          <div className='alert alert-warning'>
+            Instrument installation is not possible given the current server
+            configuration; the LORIS 'quatUser' is not configured properly.
+            File upload is still possible but instruments will need to be
+            installed manually
+          </div>
+        );
+      }
+    };
+
+    const uploadTab = () => {
+      let content = null;
+      if (this.state.data.writable) {
```

Comment:
Same concern as above.

Reply:
Same,this tells if the project/instrument and project/table_sql directory can be written into.